roofit/hs3/src/RooJSONFactoryWSTool.cxx

```diff
@@ -1584,6 +1589,70 @@ bool RooJSONFactoryWSTool::importYML(std::string const &filename)
    return this->importYML(infile);
 }
 
+bool RooJSONFactoryWSTool::importFunctionfromString(const std::string &s){
+   std::stringstream is(s);
+   std::unique_ptr<JSONTree> tree = JSONTree::create(is);
+   const JSONNode &n = tree->rootnode();
+
+   bool isDistribution = false;
+
+   // Check if function is a p.d.f. 
+   const std::string objectName = n["name"].val();
+   if(endsWith(s, "distribution")){
+      isDistribution = true;
+   }
```

Comment:
```suggestion
   // Check if function is a p.d.f. 
   bool isDistribution = endsWith(p["type"].val(), "_dist");
```
What you need to check is the name of the *type*, not the name of the object. And as you can see in for example the document I sent you, the convention is that distribution names end with `_dist`.